File: iCyption2022/practice/paket/solution.cpp
#include <bits/stdc++.h>

using namespace std;

typedef long long ll;
typedef long double ld;
const ll M = 1e9 + 7;

int main() {
    ios_base::sync_with_stdio(false);cin.tie(NULL);
    ll A, B, N; cin >> A >> B >> N;

    ll maxs = max(A, B), mins = min(A, B);
    ll numMax = N / maxs;
    while ((N - maxs * numMax) % mins != 0 && numMax > 0) {
        numMax--;
    }

    ll numMin = (N - maxs * numMax) / mins;
    if (maxs * numMax + mins * numMin == N) {
        if (maxs == A && mins == B) {
            cout << numMax << " " << numMin << endl;
            return 0;
        } else {
            cout << numMin << " " << numMax << endl;
            return 0;
        }
    }
    cout << "Pesan Ulang" << endl;
}

File: iCyption2022/practice/paket/solution.py
A, B, N = map(int, input().split())

def solve(A,B,N):
    Max = max(A,B); Min = min(A,B)
    NumMax = N // Max
    while (N - Max * NumMax) % Min != 0 and NumMax > 0:
        NumMax -= 1
    NumMin = (N - Max * NumMax) // Min
    if Max * NumMax + Min * NumMin == N:
        if Max == A and Min == B: return (NumMax,NumMin)
        else: return (NumMin,NumMax)
    else: return "Mustahil"

R = solve(A,B,N)

if R == "Mustahil": print("Pesan Ulang")
else: print(R[0],R[1],sep=' ')